src/libcore/mem.rs

```diff
@@ -209,6 +209,35 @@ pub fn forget<T>(t: T) {
 /// The mutability of a pointer does not change its size. As such, `&T` and `&mut T`
 /// have the same size. Likewise for `*const T` and `*mut T`.
 ///
+/// # Size of `#[repr(C)]` items
+///
+/// The `C` representation for items has a defined layout. With this layout,
+/// the size of items is also stable as long as all fields have a stable size.
+///
+/// ## Structs
+///
+/// For `structs`, the size is determined by the following algorithm.
+///
+/// For each field in the struct ordered by declaration order:
+///
+/// 1. Add the size of the field.
+/// 2. Round up the current size to the nearest multiple of the next field's [alignment].
+///
+/// Finally, round the size of the struct to the nearest multiple of its [alignment].
+///
+/// Unlike `C`, zero sized structs are not rounded up to one byte in size.
+///
+/// ## Enums
+///
+/// Enums that carry no data other than the descriminant have the same size as C enums
+/// on the platform they are compiled for.
+///
+/// ## Unions
```

Comment:
I don't know what's wrong with linkchecker, but it seems this title is not acceptable.

```
[01:09:19] std/mem/index.html:32: broken link fragment `#unions` pointing to `std/mem/index.html`
[01:09:25] core/mem/index.html:32: broken link fragment `#unions` pointing to `core/mem/index.html`
[01:09:28] thread 'main' panicked at 'found some broken links', /checkout/src/tools/linkchecker/main.rs:49:8
```

Reply:
@Havvy Ok the problem is that `ManuallyDrop` is a union, so there's already a "Unions" section in the documentation (check https://doc.rust-lang.org/std/mem/index.html). The linkchecker error message should say something like "duplicated link ID". 

Could we workaround by saying "Size of Structs"/"Size of Enums"/"Size of Unions"?